org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/GradleClasspathContainerUpdater.java

```diff
@@ -80,39 +81,64 @@ private void updateClasspathContainer(IProgressMonitor monitor) throws JavaModel
     }
 
     private ImmutableList<IClasspathEntry> collectClasspathContainerEntries() {
-        // project dependencies
-        List<IClasspathEntry> projectDependencies = FluentIterable.from(this.gradleProject.getProjectDependencies())
-                .transform(new Function<OmniEclipseProjectDependency, IClasspathEntry>() {
+        List<IClasspathEntry> externalDependencies = collectExternalDependencies();
+        List<IClasspathEntry> projectDependencies = collectProjectDependencies();
 
-                    @Override
-                    public IClasspathEntry apply(OmniEclipseProjectDependency dependency) {
-                        IPath path = new Path("/" + dependency.getPath());
-                        return JavaCore.newProjectEntry(path, ClasspathUtils.createAccessRules(dependency), true, ClasspathUtils.createClasspathAttributes(dependency), dependency.isExported());
-                    }
-                }).toList();
-
-        // external dependencies
-        List<IClasspathEntry> externalDependencies = FluentIterable.from(this.gradleProject.getExternalDependencies()).filter(new Predicate<OmniExternalDependency>() {
+        boolean hasExportedEntry = FluentIterable.from(externalDependencies).anyMatch(new Predicate<IClasspathEntry>() {
```

Comment:
That's unfortunately not possible atm, not without loading an additional tooling model. So far we relied on the workaround that certain features are not available in the `EclipseProject` model if loaded from an older Gradle version. I'll create a new issue for that.